src/authenticator.ts

```diff
@@ -2,26 +2,41 @@ import SessionStrategy from './strategies/session'
 import SessionManager from './session-manager'
 import { Strategy } from './strategies'
 import { FastifyRequest } from 'fastify'
+import authenticateFactory, { AuthenticateFactoryOptions } from './handlers/authenticate'
+import initializeFactory from './handlers/initialize'
+import fastifyPlugin = require('fastify-plugin')
+import { DoneCallback } from './decorators/login'
 
 type DoneFunction = (err: null | Error | 'pass', user?: any) => void
 
 // (request: FastifyRequest, user: any, done: DoneFunction): void
 
+export interface Request extends FastifyRequest {
+  _passport: {
+    instance: Authenticator
+    session: any
+  }
+  flash(key: string, message: string)
+  logIn<T = unknown>(this: Request, user: T, done: DoneCallback): void
+  logIn<T = unknown>(this: Request, user: T, options: { session?: boolean }, done?: DoneCallback): void
+  logIn<T = unknown>(this: Request, user: T, options: { session?: boolean } | DoneCallback, done?: DoneCallback)
+  authInfo: any
+  isAuthenticated(this: Request)
+}
+
```

Comment:
I am not sure that this is the best possible option. If 3rd party types are extended it is better to use type augmentation. https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation
Using augmentation will make transparent the fact that the module extends fastify's internal types: the final user will import `import { FastifyRequest } from 'fastify'` and it will be augmented with this types. If someone uses this package it will surely be augmented, so it is a desired behavior. You can find an example here: https://github.com/fastify/fastify-flash/blob/master/src/%40types/fastify.ts, however it is better to use external `typings` folder and declare an augmentation path for fastify (`./typings/fastify/index.d.ts`)

Reply:
I found a better augmentation example: https://github.com/fox1t/multer/blob/master/typings/fastify/index.d.ts

Reply:
Thank you. I‘ll take a look at it.

Reply:
I am using type augmentation now.